### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by Bloodaxe95** [Nielk1/VSCView#4]
Title; it would be awesome if us Linux users could have access to this great software as well! Looks great, and conveys information to the people watching in a good manner.

**Comment by dwecerberus** [comment]
+1 for this. I haven't tried it through Wine yet, I imagine it would probably load just fine, but getting controller support in Wine isn't always reliable, and it wouldn't be pulling directly from the Steam Controller inputs, it would be Xinput as if it were a typical Xbox controller.

Maybe [http://github.com/kozec/sc-controller](http://github.com/kozec/sc-controller) can provide some assistance or a jumping off point in getting a native Linux implementation? It already has a HUD for on-screen display, but it doesn't work like this, and definitely doesn't look as clean.

**Comment by dwecerberus** [comment]
Just an update. VSCView runs happily with Wine, but as I suspected it does not see the Steam Controller at all. I even tried using sc-controller to see if that might pass through input to VSCView but it does not. 

So any chance of getting a Linux version of this would be greatly appreciated.

**Comment by Nielk1** [comment]
I've attempted to use HidSharp instead of HidLibrary to have code that will work for Linux but I couldn't get device reading to work at all in that case without going exclusive with the controller (defeating the purpose).  The best options right now are to either alter HidLibrary to support Linux/Mono and add a Mono build configuration to VSCView or to successfully switch to the HidSharp library.

Someone with experience with HidSharp might be able to make the needed changes easily, or it might be easier to alter HidLibrary.  HidLibrary is a bit old though, so it is probably better to move away from it if possible.

**Comment by Nielk1** [comment]
Just to update, this is still not a feature.  The designed moves closer and closer though.  v0.5.0.0 now uses HidSharp fully and has a whole input layer where the HID provider could be replaced with anything.

**Comment by papamikey** [comment]
been waiting for a while for this, hopefully it gets done soon. There's currently no alternative on Linux as of now, except for SCController but it looks like it was made in 1999 and you can't really add transparency through obs because of the way it was designed

**Comment by JustCryen** [comment]
SC Controller is not an input overview software, it's a Stem Controller to xinput "driver" for supporting games outside of Steam / without Steam running in the background.

**Comment by Ryochan7** [comment]
> SC Controller is not an input overview software, it's a Stem Controller to xinput "driver" for supporting games outside of Steam / without Steam running in the background.

Pretty sure papamikey is referring to the **Input Test Mode** feature of SC Controller. That feature adds visual indicators for controller inputs on the main window to show when an input has been activated. Maybe a layer mask could be made so doing a chroma key around the controller image with OBS would be possible. It would be extra work and it would not convey some of the information that VSCView can show. Having a dedicated tool for displaying the information for use when recording might be useful.

Trying to use an app in Wine seems like a terrible approach to the problem. It would be much better to focus on working on a native Linux app for this use case. Over 4 years and nobody has stepped up to try to create some form of clone of this app.

**Comment by JustCryen** [comment]
Yeah, having something native would be much better but I'm not sure if SC Controller is a right kind of software for that.
Maybe reusing [OpenJoystickDisplay](https://github.com/KernelZechs/open-joystick-display) would be a better idea?
Profile for Steam Controller is missing and for now I was able to map Steam Controller to xbox preset only (no touch pad positions, only analog output perceived by the game)

**Comment by papamikey** [comment]
> > SC Controller is not an input overview software, it's a Stem Controller to xinput "driver" for supporting games outside of Steam / without Steam running in the background.
> 
> Pretty sure papamikey is referring to the **Input Test Mode** feature of SC Controller. That feature adds visual indicators for controller inputs on the main window to show when an input has been activated. Maybe a layer mask could be made so doing a chroma key around the controller image with OBS would be possible. It would be extra work and it would not convey some of the information that VSCView can show. Having a dedicated tool for displaying the information for use when recording might be useful.
> 
> Trying to use an app in Wine seems like a terrible approach to the problem. It would be much better to focus on working on a native Linux app for this use case. Over 4 years and nobody has stepped up to try to create some form of clone of this app.

You're right, that's what I meant, sorry if it wasn't clear. I'm about to put a 100 dollar bounty on this. I would try it but I have a job/ too busy to learn how to code.

**Comment by ShaneTRS** [comment]
Just thought I'd jot this down here since it's a jab at making a Steam Controller OSD for Linux. https://github.com/FliiFe/schud

It's not feature complete and it's powered by Electron, but it does work most of the time. I've found modifying `resources/app/src/hid.js` (line 36) to select the correct controller works well to circumvent this though.
